superset/models/core.py

```diff
@@ -722,14 +720,7 @@ def select_star(
             indent=indent, latest_partition=latest_partition, cols=cols)
 
     def wrap_sql_limit(self, sql, limit=1000):
-        qry = (
-            select('*')
-            .select_from(
-                TextAsFrom(text(sql), ['*'])
-                .alias('inner_qry'),
-            ).limit(limit)
-        )
-        return self.compile_sqla_query(qry)
+        return self.db_engine_spec.apply_limit_to_sql(sql, limit, self)
```

Comment:
Better update the method name here.

Reply:
I mean, rename `wrap_sql_limit` to something else.

Reply:
went with `apply_limit_to_sql`